### ISSUE TITLE
How to colour the label fill with gradient (amcharts5)

### ISSUE THREAD

**Issue by Ashishvkale** [amcharts/amcharts5#1636]
I'm trying to fill the label with multiple colors with the help of a linear gradient 
**Following is the code: ( non working !! )**

```
label.adapters.add("fill", (value: any, label: any) => {
                  const gradient = am5.LinearGradient.new(this.root, {
                            stops: [
                                {
                                    color: am5.color("#ff0000"), 
                                    opacity: 1,
                                    offset: 0.5
                                },
                                {
                                    color: am5.color("#ffff00"),
                                    opacity: 1,
                                    offset: 0.5
                                }
                            ],
                            rotation: 0
                        })

                        return gradient
}
```

**With help of hardcoded CSS** following is the image what I wanted to achieve 
A label with multiple colors
<img width="363" alt="Screenshot 2024-07-26 at 2 58 48 PM" src="https://github.com/user-attachments/assets/a3759744-4b10-4a27-83c8-8dad4c0ad5ec">

**Comment by martynasma** [comment]
Labels do not support gradient fills, I'm afraid.

**Comment by Ashishvkale** [comment]
@zeroin  
@martynasma Thanks for confirming the same !!

 
<img width="363" alt="Screenshot 2024-07-26 at 2 58 48 PM" src="https://github.com/user-attachments/assets/d36bb892-7d78-403b-aadd-8165f05212e7">

As we can see in above image I'm trying to color the label with hardcoded HTML 
Below is the code I'm trying to achieve the same with dynamic values : of column and label

```
label.adapters.add("html", (value, label) => {
    const columnData = label.dataItem?.dataContext as ColumnContent
    let fullTitle = ""
    // hardcoded value to show
    let overhangPixels = 20

    // column label 
    if (columnData === undefined) fullTitle = ""
    else fullTitle = columnData.columnTitle ?? ""

    if (label.dataItem?.get("graphics")) {
        // calculating colum width
        const activityWidth = label.maxWidth()
        // calculating label width
        const bounds = label.localBounds()
        const measuredWidth = bounds.right
        if (measuredWidth > activityWidth) {
            // label width in pixal which is going out of column
            overhangPixels = measuredWidth - activityWidth

            // html to add overhangPixels 
            return `<span style="background:linear-gradient(to left, #000 0px, #000 ${overhangPixels}px, #fff 20px); color: transparent; background-clip: text;">${fullTitle}</span>`
        }
    }
    
    // html with hardcoded values
    return `<span style="background:linear-gradient(to left, #000 0px, #000 ${overhangPixels}px, #fff 20px); color: transparent; background-clip: text;">${fullTitle}</span>`
})
```

But the issue here is :
HTML renders first and
till the time column and its graphics get rendered 
**label.dataItem?.get("graphics") is undefined till the time,
 so I can't calculate the width of both and can't set the dynamic multiple color value** 
 
 Is there any way to achieve the same result 
 Or how I can modify the HTML once I get the values

**Comment by martynasma** [comment]
You may attach a `dataitemchanged` event to th elabel:

```JavaScript
label.events.on("dataitemchanged", function(ev) {
  console.log(ev.target.dataItem);
  ev.target.set("html", "........");
});
```

**Comment by Ashishvkale** [comment]
Thanks @martynasma 

Not the **label.events.on("dataitemchanged",**


But the 
**label.events.on("boundschanged", (event) => {}**
worked for my scenario

**Comment by martynasma** [comment]
Requalifying it as an enhancement request. We will consider implementing this feature.

**Comment by Logikgate** [comment]
@martynasma thanks for that. Unfortunate that you cannot accept pull requests as I would love to help implement. 

For now we will rely on our fork of amcharts that contains the feature and will be on the lookout for the feature to be added to the library natively.

Thanks for everything you do to make amcharts awesome!

**Comment by martynasma** [comment]
Implemented in [5.10.1](https://github.com/amcharts/amcharts5/releases/tag/5.10.1).

## [5.10.1] - 2024-08-09

### Added
- `fillGradient` setting added to `Label`.

### Changed
- Exporting a PDF will now check image size and will not scale it up to fit page size if it's smaller. It will still scale dowwn the image to fit in the page.

### Fixed
- Memory leak with axis ranges on `LineSeries` fixed.
- `PieSeries` tick was not visible if pie had two equal slices.
- `root.nonce` was not being added to all dynamically-loaded stylesheets.
- A stack overflow could occur when auto-zooming a `ValueAxis` in some very rare cases.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by Logikgate** [comment]
@martynasma Thank you so much for this! Just tested it out and it works great
